src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -1317,6 +1317,7 @@ export class NotebookModel extends Disposable implements INotebookModel {
 							this._onValidConnectionSelected.fire(false);
 						}
 					});
+				this.sendNotebookTelemetryActionEvent(TelemetryKeys.NbTelemetryAction.ConnectionChanged);
```

Comment:
This should also include connection info - the service has a `withConnectionInfo` event for making this easy to add to events.